File: solutions/3371-harshad-number/harshad-number.cpp
class Solution {
public:
    int sumOfTheDigitsOfHarshadNumber(int x) 
    {
        int s = 0, t = x;
        while (t > 0) 
        {
            s += t % 10;
            t /= 10;
        }
        return (x % s == 0) ? s : -1;
    }
};

File: solutions/3371-harshad-number/harshad-number.java
class Solution {
    public int sumOfTheDigitsOfHarshadNumber(int x) 
    {
        int sum=0;
        int digit=x;
        while(digit>0)
        {
           int a=digit%10;
           sum+=a;
           digit/=10; 
        }
        if(x%sum==0)
        {
           return sum;
        }
        return -1;
    }
}